chef/automate
# multi region support for aws-api integration

### :nut_and_bolt: Description: What code changed, and why?
The content team has a new, certified AWS profile that supports multi-region scanning.
This code change modifies the "aws-api" integration and scan job path to support (and encourage) this updated profile/behaviour.

Previously, we were creating one node per region for every "aws-api" integration. Now that the profile itself supports multi-region scanning, we are moving to creating one node per account. We are removing the regions filter on scan job create. The inspec-agent logic has been updated to support both the old code path (one node per region) and the new one (one node per account).
We will be recommending that users delete their old "aws-api" integration and all associated nodes and create a new one to be used with the profile.

### :chains: Related Resources
https://blog.chef.io/2019/03/13/chef-software-is-the-first-cis-partner-certified-on-aws-azure-and-gcp/
https://blog.chef.io/2019/03/21/hit-the-cloud-running-with-inspec-profiles/
https://github.com/chef/automate/issues/75

### :+1: Definition of Done
- only one node created per account for "aws-api" integration
- region filters on scan job create is removed, just one node available per aws-api integration
- scans for the new style and old style both work

### :athletic_shoe: How to Build and Test the Change
- start up the services
- add an aws api integration (use okta aws functionality to get aws creds):
```
curl -s --insecure -H "api-token: $token" $url/api/v0/nodemanagers -d '{
    "name": "my aws api integration with session token",
    "type": "aws-api",
    "instance_credentials": [],
		"credential_data": [
			{"key": "AWS_ACCESS_KEY_ID", "value": "value" },
			{"key": "AWS_SECRET_ACCESS_KEY", "value": "value" },
			{"key": "AWS_SESSION_TOKEN", "value": "value" }
		]
  }'
```
- see 16 nodes added
- start a recurring job against that account (every 10 min or something)
- rebuild compliance, nodemanager, and ui
- get new creds and create another integration
- expect to see one node for this new integration
- install the new aws profile (i can't link it here b/c it's not public but if anyone needs it for testing pls dm me)
- run a scan job with new integration and new profile
- ensure scan job is created, look at the report
- go back and check on the recurring scan job running on the old integration, make sure it is still doing the right thing/executing scans and creating reports

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
<img width="1195" alt="Screen Shot 2019-08-21 at 17 50 04" src="https://user-images.githubusercontent.com/10341541/63552673-7fb34f00-c4f5-11e9-97bb-31b9908201e8.png">
<img width="1219" alt="Screen Shot 2019-08-21 at 17 50 44" src="https://user-images.githubusercontent.com/10341541/63552674-804be580-c4f5-11e9-8706-140b352d30e2.png">
<img width="1227" alt="Screen Shot 2019-08-21 at 18 08 19" src="https://user-images.githubusercontent.com/10341541/63552675-804be580-c4f5-11e9-9269-ba946b3b4e20.png">
